ansible/slurm-drain.yml

```diff
@@ -0,0 +1,38 @@
+---
+- import_playbook: cluster-infra-query.yml
+
+- hosts: openstack
+  gather_facts: false
+  tasks:
+    - name: Count the number of compute nodes per slurm partition
+      set_fact:
+        desired_state: "{{ ((cluster_group.output_value | selectattr('group', 'equalto', item.name ) | first).nodes | map(attribute='name') | list )[:item.num_nodes] + desired_state | default([]) }}"
```

Comment:
Does this work if an expected partition is not found in the running stack?  Might be worth checking for that first if not.  Similarly, if an extra partition is unexpectedly found

Reply:
It did fail but I have address this in a new commit.